core/src/test/scala/integration/kafka/admin/TopicCommandIntegrationTest.scala

```diff
@@ -586,11 +586,14 @@ class TopicCommandIntegrationTest extends KafkaServerTestHarness with Logging wi
     try {
       killBroker(0)
       val aliveServers = brokers.filterNot(_.config.brokerId == 0)
-      TestUtils.waitForPartitionMetadata(aliveServers, testTopicName, 0)
+      TestUtils.waitUntilTrue(
```

Comment:
Can we alternatively use one of the existing methods in TestUtils to validate that the topic partition ISR contains rest of the 5 brokers e.g.
using `TestUtils.waitForBrokersInIsr` could validate that the topic partition metadata exists in expected number of Isr even after one of the brokers is terminated.

Reply:
@divijvaidya I really appreciate your suggestion, however, I need to clarify something here: 
1. We should use `TestUtils.waitForBrokersOutOfIsr` not `TestUtils.waitForBrokersInIsr`
2. `TestUtils.waitForBrokersInIsr` needs to send RPC but we can directly use MetadataCache which is in the memory, so it's faster and more reliable, maybe we can improve `TestUtils.waitForBrokersInIsr`  in the future
3. This is most important, in another PR we found we can use a more generic way in KRaft mode, which waits until the brokers have caught up to the controller metadata topic end offset, instead of waiting for something really specific.